config/v2/resolvers.go

```diff
@@ -316,13 +283,35 @@ func AWSAccountsNeededGetter(comm Common) *bool {
 }
 
 func ResolveAWSAccountsNeeded(commons ...Common) bool {
-	accountsNeeded := lastNonNilBool(AWSAccountsNeededGetter, commons...)
+	accountsNeeded := lastNonNil(AWSAccountsNeededGetter, commons...)
 	if accountsNeeded == nil {
 		return true
 	}
 	return *accountsNeeded
 }
 
+func ExtraTemplatesGetter(comm Common) *[]ExtraTemplate {
+	if comm.ExtraTemplates != nil {
+		templates := []ExtraTemplate{}
+		for _, v := range *comm.ExtraTemplates {
+			templates = append(templates, ExtraTemplate{
+				Overwrite: v.Overwrite,
+				Files:     v.Files,
+			})
+		}
+		return &templates
```

Comment:
Potential opportunity to make a helper to consolidate same mapping functionality (1)

Reply:
you're right, but unfortunately, the way this codebase's packages are laid out makes this difficult without doing a bit of refactoring. The config stage structs are all pointers because the configuration can or cannot have values and then they duplicate these same structs in the plan phase once they are "resolved" (going down the tree of overwrites from global -> default -> account -> env -> component) through the tree overwrites. Resolved structs don't have pointers. I would love to simplify this code since its so copy/pasted, but maybe I will try that in a different PR